app/screens/home/channel_list/servers/servers_list/server_item/index.tsx

```diff
@@ -0,0 +1,257 @@
+        borderRadius: 8,
+        flexDirection: 'row',
+        height: 72,
+        marginBottom: 12,
+    },
+    details: {
+        marginLeft: 14,
+    },
+    logout: {
+        backgroundColor: theme.centerChannelBg,
+        borderRadius: 8,
+        height: 16,
+        left: 40,
+        position: 'absolute',
+        top: 11,
+        width: 16,
+    },
+    nameContainer: {
+        alignItems: 'center',
+        flexDirection: 'row',
+    },
+    name: {
+        color: theme.centerChannelColor,
+        ...typography('Body', 200, 'SemiBold'),
+    },
+    offline: {
+        opacity: 0.5,
+    },
+    row: {flexDirection: 'row'},
+    unread: {
+        top: -2,
+        left: 25,
+    },
+    url: {
+        color: changeOpacity(theme.centerChannelColor, 0.72),
+        ...typography('Body', 75, 'Regular'),
+    },
+    websocket: {
+        marginLeft: 7,
+        marginTop: 4,
+    },
+}));
+
+const ServerItem = ({isActive, server}: Props) => {
+    const intl = useIntl();
+    const theme = useTheme();
+    const [badge, setBadge] = useState<BadgeValues>({isUnread: false, mentions: 0});
+    const styles = getStyleSheet(theme);
+    const swipeable = useRef<Swipeable>();
+    const subscription = useRef<Subscription|undefined>();
+    const websocketError = server.lastActiveAt ? !WebsocketManager.isConnected(server.url) : false;
+    let displayName = server.displayName;
+    if (server.url === server.displayName) {
+        displayName = intl.formatMessage({id: 'servers.default', defaultMessage: 'Default Server'});
+    }
+
+    const unreadsSubscription = (myChannels: MyChannelModel[]) => {
+        let mentions = 0;
+        let isUnread = false;
+        for (const myChannel of myChannels) {
+            mentions += myChannel.mentionsCount;
+            isUnread = isUnread || myChannel.isUnread;
+        }
+
+        setBadge({isUnread, mentions});
+    };
+
+    const containerStyle = useMemo(() => {
+        const style = [styles.container];
+        if (isActive) {
+            style.push(styles.active);
+        }
+
+        return style;
+    }, [isActive]);
+
+    const serverStyle = useMemo(() => {
+        const style = [styles.row];
+        if (!server.lastActiveAt) {
+            style.push(styles.offline);
+        }
+
+        return style;
+    }, [server.lastActiveAt]);
+
+    const onServerPressed = useCallback(() => {
+        if (isActive) {
+            // eslint-disable-next-line no-console
+            console.log('ACTIVE SERVER', server.displayName);
+            DeviceEventEmitter.emit(Navigation.NAVIGATION_CLOSE_MODAL);
+            return;
+        }
+
+        if (server.lastActiveAt) {
+            // eslint-disable-next-line no-console
+            console.log('SWITCH TO SERVER', server.displayName);
+            return;
+        }
+
+        // eslint-disable-next-line no-console
+        console.log('LOGIN TO SERVER', server.displayName);
+    }, [server]);
```

Comment:
Missed isActive in dependency array?

Reply:
it actually is missing all functionality as well :P